packages/commerce-sdk-react/src/hooks/useDNT.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import useAuthContext from './useAuthContext'
+import useConfig from './useConfig'
+
+interface useDntReturn {
+    dntStatus: boolean | undefined
+    updateDNT: (preference: boolean | null) => Promise<void>
+}
+
+/**
+ * Hook that returns
+ * dntStatus - a boolean indicating the current DNT preference
+ * updateDNT - a function that takes a DNT preference and creates the dw_dnt
+ *              cookie and reauthroizes with SLAS
+ *
+ * @group Helpers
+ * @category DNT
+ *
+ */
+const useDNT = (): useDntReturn => {
+    const auth = useAuthContext()
+    const dwDntValue = auth.getDnt()
+    const updateDNT = async (preference: boolean | null) => {
+        await auth.setDnt(preference)
+    }
+
+    let dntStatus = undefined
```

Comment:
Why do we need to hold another variable for dntValue in this hook? Can we safely rely on `auth.getDnt()`? I would image when the user logs out, it should be possible to remove the cookie. That means the dntVal at that time can also be `undefined`

Reply:
There is also a possibility that the user manually changed the cookie value to be something invalid (a value neither '1' or '0'), which is the fourth case after 'undefined'. We would still interpret this to be an `undefined` dntStatus, so we need another variable to take that into account.

Reply:
Can it be achieve that in getDNT cookies? since we are only sycing the cookies on refresh, meaning getDNT is getting called, if the DntVal is invalid, we remove it. 🤔

Reply:
Yes this can be done in getDNT. Updated!